### DISCUSSION TITLE
Hard-coded mock responses with Mockoon

### DISCUSSION THREAD

**Discussion by jnimander** [mockoon/mockoon#904] [Help] [answered]
I'm trying to mock an API like this,

`GET users/:userID`

Then create hard-coded (not templated or faked) responses using data buckets. My data bucket looks like this.

```
[
  {
    "{{urlParam 'userId'}}": "10000",
    "firstname": "John",
    "lastname": "Doe",
    "friends": []
  },
  {
    "{{urlParam 'userId'}}": "10001",
    "firstname": "Archie",
    "lastname": "Bunker",
    "friends": []
  }
]
```

Mockoon always returns the entire data bucket contents. Is there any way for me to match the urlParam to an item in an array in my data bucket?

**Comment by 255kb** [comment, marked as answer]
This is not yet possible but will be part of the CRUD endpoints feature (#124). It will behave like you expect and use the data buckets. 
Currently, Mockoon's endpoints are all independant and stateless. Except for the databuckets that are generated once when the server start.

**Reply by jnimander** [reply]
OK, thanks for the info. Looking forward to getting this feature as it's kind of a must for me. Otherwise an awesome product.

**Reply by 255kb** [reply]
Thanks :) 
I think it's the most awaited feature!

**Reply by famblard86** [reply]
Yes, it's a awesome project.
I confirm I would really like a way to make a simple CRUD with share data buckets in multi endpoints.
Did you have some idea about it, or some workaround to do it in the actual version ?

**Reply by 255kb** [reply]
The implementation will be similar to what json-server does. 
And currently, I don't think there is a very practical alternative that doesn't involve hard coding IDs in the rules.
